### PR TITLE
[#1053] Adding tests for cloudbees folders plugin

### PR DESCRIPTION
**Pull request by rsandell**
See #1053 

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/plugins/CloudbeesFoldersTest.java
Comment (jtnord) [inline src/test/java/plugins/CloudbeesFoldersTest.java:1]: how does this relate to https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/FolderPluginTest.java

whilst the canonical name should be as in this file - how much overlap is there between the two?  (can we have just one that is named appropriately?)

Comment (rsandell) [reply src/test/java/plugins/CloudbeesFoldersTest.java:1]: Weird, I might need new glasses :sweat_smile: , I looked for existing tests before adding this but couldn't find any. I'll take another look to make sure there is no overlap.

Comment (rsandell) [reply src/test/java/plugins/CloudbeesFoldersTest.java:1]: The tests are testing different things and in different scenarios.
Question though if I should merge them into one class or keep them separated. The existing tests are "simpler" and extending the "base" JUnit test class while the new ones are using another for some more testing abilities.

Comment (jtnord) [reply src/test/java/plugins/CloudbeesFoldersTest.java:1]: https://github.com/jenkinsci/acceptance-test-harness/pull/1092/files#diff-6d25650c00d8df0d73c417ca9820dcfde553788283acc8842ce5e0571658ff02R130-R148 looks suspiciously like https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/FolderPluginTest.java#L118-L140

I took a quick look at the code nothing stood out as needing to come from the different base class? only one test seemed to do anything pipeline releated and did not appear to use the new base class?

**Review by ampuscas — APPROVED** [review]

**Review by olamy — APPROVED** [review]

File: src/test/java/plugins/FolderPluginTest.java [hunk 1]
```diff
@@ -24,24 +24,31 @@
...
 
+import org.apache.commons.io.IOUtils;
 import org.jenkinsci.test.acceptance.Matchers;
 import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
 import org.jenkinsci.test.acceptance.junit.WithPlugins;
 import org.jenkinsci.test.acceptance.plugins.credentials.CredentialsPage;
 import org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials;
 import org.jenkinsci.test.acceptance.plugins.credentials.UserPwdCredential;
 import org.jenkinsci.test.acceptance.po.*;
+import org.junit.Assert;
```
Comment (NotMyFault) [inline src/test/java/plugins/FolderPluginTest.java:35]:
```suggestion
```
The pre-existing tests already use org.hamcrest.MatcherAssert#assertThat(), let's stick to one #assertThat.

File: src/test/java/plugins/FolderPluginTest.java [hunk 2]
```diff
@@ -167,6 +178,144 @@ public void folderScopedCredentialsTest() {
...
+
+            Assert.assertThat("Complete path for item is not displayed", driver, hasContent(itemPath));
+        }
+    }
+
+    private void checkItemExists(final TopLevelItem f, final boolean expectedExists) {
+        boolean exists = true;
+
+        try {
+            IOUtils.toString(f.url("").openStream());
```
Comment (NotMyFault) [inline src/test/java/plugins/FolderPluginTest.java:311]:
```suggestion
            IOUtils.toString(f.url("").openStream(), StandardCharsets.UTF_8);
```
If applicable (import needed), if not, feel free to disregard 👏

**Review by car-roll — APPROVED** [review]

**Pull request merged**